src/classpath-openjdk.cpp

```diff
@@ -972,37 +984,19 @@ class EmbeddedFile {
   unsigned pathLength;
 };
 
-void uncaughtException(Thread *t, GcThrowable *e)
+static uint64_t uncaughtExceptionHandler(Thread* t, uintptr_t* arguments)
 {
   GcMethod* dispatch = resolveMethod(t,
                                      roots(t)->bootLoader(),
                                      "java/lang/Thread",
                                      "dispatchUncaughtException",
                                      "(Ljava/lang/Throwable;)V");
   if (dispatch != NULL) {
-    THREAD_RESOURCE0(t, {
-      if (t->exception != NULL) {
-        // The stack will be unwound when this resource is
-        // released, which means that uncaughtException()
-        // will not return. So repeat the thread clean-up here.
-        disposeThread(t);
-      }
-    });
-
-    t->m->processor->invoke(t, dispatch, t->javaThread, e);
+    GcThrowable* exception
```

Comment:
We need to move this to before the call to resolveMethod and protect it:
```c++
GcThrowable* exception = cast<GcThrowable>(t, reinterpret_cast<object>(arguments[0]));
PROTECT(t, exception);
``` 
Otherwise, `resolveMethod` may trigger a GC and move the object `arguments[0]` points to without updating the pointer.